### PR TITLE
Simon/issue#11/error handling

### PR DESCRIPTION
**Pull request by psi77**
Some error handling and an error page and some tests!

### CODE REVIEW

File: src/app/obz.service.ts [hunk 1]
```diff
@@ -72,123 +61,115 @@ export class ObzService {
...
-            });
-          }));
-        });
-      }
-      Promise.all(promises).then(() => {
-        observer.complete();
-      });
+  private parseImage = (zip, image: string, boardSet: OBZBoardSet): Promise<void> => {
+    const encoding = image.toLowerCase().endsWith('.svg') ? 'text' : 'base64';
+    const ifl = zip.file(image);
```
Comment (blaise17) [inline src/app/obz.service.ts]: Please can this, sf and bf all be made into readbale names :)

File: src/karma-obz-preprocessor/karma-obz-preprocessor.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+var createOBZPreprocessor = function (logger, basePath, config) {
+
+  var testBasePath = basePath.replace('/src', '');
+  var log = logger.create('preprocessor.obz');
+  var ret = function (content, file, done) {
+
+    var fixtureName = file.originalPath
+      .replace(testBasePath + '/', '')
+      .replace('test/fixtures/obz/', '')
+      .replace(/\.obz$/, '');
```
Comment (blaise17) [inline src/karma-obz-preprocessor/karma-obz-preprocessor.js]: How terribly unpleasant. Thank you for taking this one for the team :)

File: src/test-utils/OBZFixture.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+export class OBZFixture {
+
+  constructor() {
+  }
+
+  static load = (fixtureName) => {
+    // pull fixture from window.__obz__ array
+    const fixtures = window['__obz__'];
+    const noManifest = fixtures[fixtureName];
```
Comment (blaise17) [inline src/test-utils/OBZFixture.ts]: if this is for more than just noManifest can this variable please be renamed to something more generic?

File: src/app/obz.service.ts [hunk 4]
```diff
@@ -72,123 +61,115 @@ export class ObzService {
...
-              });
-            }
-          });
+    this.getOBZFile(boardURL).subscribe(
+      (blob: Blob) => {
+        this.parseOBZFile(blob).then(boardSet => {
+          this.observer.next(boardSet);
+        }).catch(error => {
+          // error loading zip file
+          throw new FatalOpenVoiceFactoryError(ErrorCodes.OBZ_PARSE_ERROR, `Could not parse ${boardURL} as a zip file`, error);
```
Comment (blaise17) [inline src/app/obz.service.ts]: Nice errors :)

File: src/app/obz.service.spec.ts [hunk 5]
```diff
@@ -14,4 +16,104 @@ describe('ObzService', () => {
   it('should be created', inject([ObzService], (service: ObzService) => {
     expect(service).toBeTruthy();
   }));
+
+  it('getBoardSet should throw fatal error if unable to load file', (done) => {
```
Comment (blaise17) [inline src/app/obz.service.spec.ts]: NOM NOM NOM TESTS!!! :D

File: src/app/obfpage/obfpage.component.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
```
Comment (blaise17) [inline src/app/obfpage/obfpage.component.ts]: Pretty :)

File: src/app/obz.service.ts (hunk 1 above)
Comment (psi77) [reply src/app/obz.service.ts]: I'm going to assume you mean meaningful and don't just wont iffle, biffle and siffle

**Pull request merged**